nipype/algorithms/tests/test_auto_ACompCor.py

```diff
@@ -22,7 +22,10 @@ def test_ACompCor_inputs():
         ),
         num_components=dict(xor=['variance_threshold'], ),
         pre_filter=dict(usedefault=True, ),
-        realigned_file=dict(mandatory=True, ),
+        realigned_file=dict(
+            extensions=None,
```

Comment:
This is an oversight from the previous PR - https://github.com/nipy/nipype/pull/2962/commits/6c3858ab81c6d375a53050cb89c896f94c86b916#diff-07934241447dd5e65f51347e36fdd1ffR291

Reply:
Right. But, for example, `exists=False` is not in all of these, even though that's the default. Why is `extensions=<default>` in here? Is this something we can fix in the trait, instead?

Reply:
We could (and it was like that before that commit) replicate the same behavior of `exists`. Namely, the metadata is not propagated up to the base classes. I'm clueless about the implications of this.

For `exists` I just kept the original behavior.

Reply:
i think the intent was not so much to track the defaults of the class, but keywords that the user put in explicitly.

i do agree that these underlying metadata (if changed internally in traits_extension) will change the behavior of interfaces and it would be good to catch them.  another side note, is how we deal with and improve pickled crashfile compatibility.

if we keep old behavior, then we must commit to not changing underlying defaults in traits_extension in the future.

Reply:
@satra Just to clarify, adding `extensions` metadata to everything is preferable?

Reply:
@effigies - if we do want to take note of underlying metadata changes, we either write a set of tests for each trait type that's used in nipype, and not add the defaults to these auto-tests.

or we simply pull every default into the auto test. more generally though, my hope is that we can move closer to the kind of testing we are planning to do with the afni folks. also start splitting interfaces into their own packages. 

so for now, i think this ok. but we should think this through/generalize our goals a bit better.